Mjml.Net/Includes/Filesystem/FilesystemFileLoader.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Text;
+using Mjml.Net.Components;
+
+namespace Mjml.Net.Includes.Filesystem;
+
+/// <summary>
+/// Provides files from a local filesystem.
+/// </summary>
+public class FilesystemFileLoader : CacheableFileLoader
```

Comment:
Your propositions?

Reply:
I would just put that in a few abstract classes, I think it is very unlikely that someone needs another file loader or so and can still do that by inheritance.

Reply:
In my opinion, this is a correct design.

`FilesystemFileLoader` inherits from `CacheableFileLoader` because it wants to have caching built-in. I decided to perform it via inheritance because we have nothing to do with composition here. If you want to make it with composition - we need to create some interception behavior here so cache interceptor will run before and after LoadText method. I don't think we need to overcomplicate that much here.

In other hand, we have two composition elements - `pathResolver` and `pathEqualityComparer`.

Regarding `pathEqualityComparer` - in .NET equality comparer can be passed as a parameter to every collection constructor or any comparison or searching mechanism, so I think it's common practice to have it, especially when it comes to the caching. Since we most likely want to have a caching built-in here, this thing is a parameter here to be passed for caching mechanism, that's it.

Regarding `pathResolver` - you right, most users won't need it for `FilesystemFileLoader`, but I don't see a reason to cut the flexibility of this loader. There is a chance that not in all cases user will be satisfied with default path resolving behavior even for filesystem. I think it causes no problem to have this optional argument for anyone. In any case, `CacheableFileLoader` requires path resolver to operate correctly, so all we can do here is to remove `pathResolver` argument from constructor and pass `new FilesystemPathResolver(baseFolder)` to base class. So why should we cut the flexibility here?